src/target/ext/host_io.rs

```diff
@@ -251,27 +251,27 @@ define_ext!(HostIoOps, HostIo);
 /// Nested Target Extension - Host I/O open operation.
 pub trait HostIoOpen: HostIo {
     /// Open a file at filename and return a file descriptor for it, or return
-    /// -1 if an error occurs.
+    /// [`HostIoError::Errno`] if an error occurs.
     ///
     /// The filename is a string, flags is an integer indicating a mask of open
     /// flags (see [`HostIoOpenFlags`]), and mode is an integer indicating a
     /// mask of mode bits to use if the file is created (see
-    /// [`HostIoMode`]).
+    /// [`HostIoOpenMode`]).
     fn open(
         &mut self,
         filename: &[u8],
         flags: HostIoOpenFlags,
-        mode: HostIoMode,
+        mode: HostIoOpenMode,
     ) -> HostIoResult<u32, Self>;
 }
 
 define_ext!(HostIoOpenOps, HostIoOpen);
 
 /// Nested Target Extension - Host I/O close operation.
 pub trait HostIoClose: HostIo {
-    /// Close the open file corresponding to fd and return 0, or -1 if an error
-    /// occurs.
-    fn close(&mut self, fd: i32) -> HostIoResult<u32, Self>;
+    /// Close the open file corresponding to fd and return 0, or
+    /// [`HostIoError::Errno`] if an error occurs.
+    fn close(&mut self, fd: u32) -> HostIoResult<u32, Self>;
```

Comment:
see my comment about setfs wrt "return 0 on success". the same thing applies to close